### ISSUE TITLE
Ensure every form element has a label (input)

### ISSUE THREAD

**Issue by azaslonov** [paperbits/paperbits-demo#110]
User Experience:

If label is not associated with edit field then screen reader user will not be able to find out what kind of information is needed in the edit fields and will face difficulties while navigating on the page.

Note: User credentials should NOT be included in the bug.

 

Repro Steps:

1. Open Azure portal with valid credentials and create a API Management.

2. Click Developer Portal button to navigate to the portal.

3. Wait portal to load in design mode

4. Navigate to the Left nav and select "Styles" button.

5. Navigate to the various controls of the "Styles" window.

6. Run ‘Accessibility insights for web’ tool and select ‘Fast Pass’ option.

7. Verify the issue.

 

Actual Result:

Label is not associated with input field present below forms controls.

 

Expected Result:

Label should be associated with input fields, present below forms controls.

 

Issue: Ensures every form element has a label (label - https://dequeuniversity.com/rules/axe/3.3/label?application=msftAI)

 

Target application: API Management - https://accessibility.developer.azure-api.net/

 

Element path: iframe;input

 

Snippet: <input type="text" class="form-control no-pointer-events form-control-default" value="Input value">

 

How to fix:

Fix any of the following:

  aria-label attribute does not exist or is empty

  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty

  Form element does not have an implicit (wrapped) <label>

  Form element does not have an explicit <label>

  Element has no title attribute or the title attribute is empty

 

This accessibility issue was found using Accessibility Insights for Web 2.14.1 (axe-core 3.3.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.

 

MAS Reference:

MAS 3.3.2 - Labels or Instructions (19)

 

Reference Links

Accessibility Insights! - Identify accessibility bugs before check-in and make bug fixing faster and easier.
External Bug Process: If this bug belongs to external team, mark it as resolved/done and assign it back to the tester with notes on where to file/route the bug. For more information please use this Link. “HCL Staff should not log any third party external bugs. Should be routed to edad team”.
Please reach out to C&AI Teams channel for any process related queries.
 

![image](https://user-images.githubusercontent.com/2320302/78179754-9409e380-7416-11ea-9ce7-cfedf1c0d547.png)

**Comment by azaslonov** [comment]
Fixed.
